src/transformers/trainer.py

```diff
@@ -2553,6 +2554,9 @@ def hyperparameter_search(
             direction (`str`, *optional*, defaults to `"minimize"`):
                 Whether to optimize greater or lower objects. Can be `"minimize"` or `"maximize"`, you should pick
                 `"minimize"` when optimizing the validation loss, `"maximize"` when optimizing one or several metrics.
+            directions (`List[str]`, *optional*, defaults to None):
+                Whether to optimize multi objectives. Can be List of `"minimize"` and `"maximize"`, you should pick
+                `"minimize"` when optimizing the validation loss, `"maximize"` when optimizing one or several metrics.
```

Comment:
Why do we add directions here? Couldn't we allow `direction` to be a string or list of strings?

Reply:
aligned with https://github.com/optuna/optuna/blob/7b3824aaab38843fdfbfc67ea93d6da446dc1548/optuna/study/study.py#L1125

Reply:
done

Reply:
change direction to be a string or list of strings?